src/main/webapp/script.js

```diff
@@ -83,13 +82,29 @@ function schedule() {
   [startDate, endDate] = daterange.split(' - ');
   formContent.append("startDate", startDate.trim());
   formContent.append("endDate", endDate.trim());
+  appendDurations(formContent);
   postData("/schedule", new URLSearchParams(formContent).toString())
       .then(getJsonIfOk)
       .then(updateView)
       .then(refreshCalendar)
       .catch(handleNetworkError);
 }
 
+/**
+ * Appends to the form data the selected values for the duration of each task.
+ */
+function appendDurations(formData) {
+  $("input[name=taskId]").each(
+      function () {
+        if (this.checked) {
+          const durationSelect = $(this).closest(".task-title").next()[0];
+          const mdcDurationSelect = new mdc.select.MDCSelect(durationSelect);
```

Comment:
This is not advisable to create a new MDCSelect. You have already created one in `mdc.select.MDCSelect.attachTo`.
I've verified that they create a new instance every time, what mean that probably event listeners etc are attached once again - needlessly.
Instead, when creating the MDCSelect, store them somewhere for future reference (in some taskId->select map, or as custom property on the DOM node)